test/react.test.js

```diff
@@ -0,0 +1,222 @@
+    tourButtons.find('.tour_button').simulate('click');
+    expect(wrapper.state('listView')).toEqual(true);
+    tourButtons.find('.tour_button').simulate('click');
+
+  });
+
+
+});
+
+describe('Introduction', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <Introduction />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    });
+  }, 1000);
+
+});
+
+
+describe('Picture', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <Picture />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    }, 1000);
+
+  });
+
+  it('Picture component should render the picture div', () => {
+    setTimeout(() => {
+      expect(shallow(<Picture />).is('.picture')).toBe(true);
+    }, 1000);
+  });
+
+});
+
+describe('Save', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <Save />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    });
+  }, 1000);
+
+  it('Save component should render the save button', () => {
+    setTimeout(() => {
+      expect(shallow(<Picture />).is('.save_button')).toBe(true);
+    }, 1000);
+  });
+
+});
+
+describe('Share', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <Share />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    });
+  }, 1000);
+
+  it('Share component should render the share button', () => {
+    setTimeout(() => {
+      expect(shallow(<Picture />).is('.share_button')).toBe(true);
+    }, 1000);
+  });
+
+});
+
+describe('Tour', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <Tour />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    });
+  }, 1000);
+
+  it('Tour component should render the tour button', () => {
+    setTimeout(() => {
+      expect(shallow(<Picture />).is('.tour_button')).toBe(true);
+    }, 1000);
+  });
+
+});
+
+describe('TourPage', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <TourPage />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    });
+  }, 1000);
+
+  it('TourPage component should render the pictures', () => {
+    setTimeout(() => {
+      expect(shallow(<TourPage />).is('.picture')).toBe(true);
+    }, 1000);
+  });
+
+});
+
+describe('ListView', () => {
+
+  it('renders correctly', () => {
+    setTimeout(() => {
+      const rendered = renderer.create(
+        <ListView />
+      );
+      expect(rendered.toJSON()).toMatchSnapshot();
+    });
+  }, 1000);
+
+  it('ListView component should render the pictures', () => {
+    setTimeout(() => {
+      expect(shallow(<ListView />).is('.picture')).toBe(true);
+    }, 1000);
+  });
+
+});
```

Comment:
These tests look lovely. I'm still pretty murky on how jest works, so unfortunately I'm not the guy to review the functionality of these for you. I did look over the file and didn't see anything that a linter wouldn't have liked.